R/dsl.R

```diff
@@ -39,5 +39,18 @@ mcstate_dsl <- function(x, type = NULL) {
     x <- rlang::quo_get_expr(quo)
   }
   exprs <- dsl_preprocess(x, type)
+  dat <- dsl_parse(exprs)
   NULL
 }
+
+
+mcstate_dsl_parse <- function(x, type = NULL) {
+  quo <- rlang::enquo(x)
+  if (rlang::quo_is_symbol(quo)) {
+    x <- rlang::eval_tidy(quo)
+  } else {
+    x <- rlang::quo_get_expr(quo)
+  }
+  exprs <- dsl_preprocess(x, type)
+  dsl_parse(exprs)
+}
```

Comment:
What's the difference between mcstate_dsl and mcstate_dsl_parse ?

Reply:
This one eventually will be something fast to run to parse the expressions and return some metadata but not do any generation; we use that pattern in odin.  It's not really used yet, but will eventually get exported